server/src/main/java/org/opensearch/monitor/fs/FsInfo.java

```diff
@@ -91,6 +94,10 @@ public Path(StreamInput in) throws IOException {
             total = in.readLong();
             free = in.readLong();
             available = in.readLong();
+            if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
Let's gate these behind the feature flag as well. Until we've finalized things and moved it out of experimental, we don't want to introduce any compatibility issues with instances that did not enable the feature flag.

Reply:
Sure. Let me update that.